libkatana_python_native/include/katana/python/CythonIntegration.h

```diff
@@ -151,16 +153,20 @@ DefCythonSupport(pybind11::class_<T, Extra...> cls) {
   // keep_alive<0, 2> causes the owner argument to be kept alive as long as the
   // returned object exists. See
   // https://pybind11.readthedocs.io/en/stable/advanced/functions.html#keep-alive
-  cls.template def_static(
-      "_make_from_address_shared", [](uintptr_t addr) -> std::shared_ptr<T> {
-        auto* ptr = reinterpret_cast<std::shared_ptr<T>*>(addr);
-        return *ptr;
-      });
-  cls.template def_static(
-      "_make_from_address_unique", [](uintptr_t addr) -> std::unique_ptr<T> {
-        auto* ptr = reinterpret_cast<std::unique_ptr<T>*>(addr);
-        return std::move(*ptr);
-      });
+  if constexpr (std::is_same_v<Holder, std::shared_ptr<T>>) {
+    cls.template def_static(
```

Comment:
What's going on here exactly?

Reply:
I'm only adding this method if Holder is `shared_ptr`. This prevents creating objects that will provide lifetime issues since pybind11 is not dynamically flexible about this and has no static or runtime checking.